activejob/lib/active_job/serializers/duration_serializer.rb

```diff
@@ -4,12 +4,20 @@ module ActiveJob
   module Serializers
     class DurationSerializer < ObjectSerializer # :nodoc:
       def serialize(duration)
-        super("value" => duration.value, "parts" => Arguments.serialize(duration.parts))
+        super("value" => duration.value, "parts" => Arguments.serialize([duration.parts]))
       end
 
       def deserialize(hash)
         value = hash["value"]
-        parts = Arguments.deserialize(hash["parts"])
+        deserialized_parts = Arguments.deserialize(hash["parts"])
+        parts = if deserialized_parts.first.is_a?(Hash)
+          deserialized_parts.first
+        else
+          # 'parts' used to be serialized like [:years, 1], but it should have been [{ years: 1 }].
+          # This maintains deserialization-compatibility with any old serialized data.
+          # TODO: remove in 7.1
```

Comment:
I think main is 7.1, and the comment was from January 2022, which I think was before that decision. So probably should change the comment at least?

Reply:
“Remove before 7.2” ?

Reply:
I think I’d originally expected this to make it into a 7.0 patch release, but if you don’t expect another patch release before 7.1 then bumping the comment makes sense